include/reference_wrapper_helper.h

```diff
@@ -312,4 +313,64 @@ namespace quda
     return out;
   }
 
+  /**
+     Derived specializaton of std::vector<T>
+     which allows us to write generic multi-scalar functions.
+   */
+  template <class T> struct vector : public std::vector<T> {
```

Comment:
I'm not sure that I love calling this class `vector`, which (namespace aside) directly conflicts with a C++ container (yes, I know we have the same issue with `complex`, that's a whole other can of worms). Another reason I don't love it is because we have `using vector = ...` higher up in this file, which could lead to confusion.

All that said, I don't necessarily have a suggestion for a better name (right now)...

Reply:
The definition above is just a class-wide convenience wrapper, but it's something that could easily be changed if it makes you happier 😄 

I did think about other names, but decided `vector` is the correct name.  It is derived from `std::vector` after all, just with some convenience additions.

I would argue that it doesn't conflict with `std::vector` since we never have a `using namespace std;` statement in QUDA.

Reply:
Maybe `vector_scalar`, but it's not necessarily a hill I'll die on... I appreciate that `vector` is the best/correct name, it's just unfortunate that it coincides with the name of a `std` container. Such is life.

Reply:
If it makes you feel any better, `cvector` is the type that is used in all interfaces, and the only time when `vector` is used in actual user code, per se, is when we need to create an explicit lvalue object.  All other uses are implicit (and `cvector`).